src/app/utils/pinecone.ts

```diff
@@ -1,13 +1,12 @@
-import { PineconeClient, ScoredVector } from "@pinecone-database/pinecone";
+import { Pinecone, ScoredVector } from "@pinecone-database/pinecone";
 
-let pinecone: PineconeClient | null = null;
+let pinecone: Pinecone | null = null;
 
 export const getPineconeClient = async () => {
```

Comment:
This `getPineconeClient` function isn't really needed anymore. In the past we had these wrapper functions mainly because of the awkward `await pineconeClient.init()` thing that was needed, and also to cache the result of the init() side effect (a call to `/whoami`) so we're not firing off a bunch of extra calls every time we use the client. But there's no init step any more, and the results of the call are cached in a singleton (projectIdSingleton) regardless of how many client instances are created.

You can just do `const pinecone = new Pinecone()` wherever you need a client instance and it will read the `PINECONE_API_KEY` and `PINECONE_ENVIRONMENT` keys from the environment variables without you having to do anything extra. You can read them yourself and pass them in to the constructor as a config object if that's what you prefer, but it's not necessary. The relevant source code is [here](https://github.com/pinecone-io/pinecone-ts-client/blob/main/src/pinecone.ts#L49).

Reply:
Makes sense! Handled in 14650f5.